### DISCUSSION TITLE
types.ParseDateTime should support int64 and unsigned?

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#2602] [Q&A]
Hi, 

I was using int64 to represent my struct's unix time value field, and noticed that `types.ParseDateTime` does not support it: https://github.com/pocketbase/pocketbase/blob/master/tools/types/datetime.go#LL82C8-L82C8

The `cast.ToTime` method perfectly handles it, it actually casts int's to int64's itself. Could `types.ParseDateTime` support it too?

Thanks,

**Comment by ganigeorgiev** [comment]
I'll consider updating it with the next release, but the recommended type when submitting `date` field values is string in the format `Y-m-d H:i:s.uZ` because this will be how it is stored in the DB. The search filter also doesn't perform casting on the operands and the string format needs to be used.

**Reply by pnmcosta** [reply]
Yup got it, but this is a source value from a third party that I need to normalize into a pb collection record field, for which I am using `types.DateTime` in my custom struct.

**Reply by pnmcosta** [reply]
FYI, using this example for now, but it's also cause it's in milliseconds `date, err := types.ParseDateTime(time.Unix(0, entry.Time*int64(time.Millisecond)))`
